src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -898,33 +898,40 @@ InboundLedger::takeTxNode(
         return true;
     }
 
-    auto nodeIDit = nodeIDs.cbegin();
-    auto nodeDatait = data.begin();
-    TransactionStateSF filter(
-        mLedger->txMap().family().db(), app_.getLedgerMaster());
-
-    while (nodeIDit != nodeIDs.cend())
+    try
     {
-        if (nodeIDit->isRoot())
-        {
-            san += mLedger->txMap().addRootNode(
-                SHAMapHash{mLedger->info().txHash},
-                makeSlice(*nodeDatait),
-                snfWIRE,
-                &filter);
-            if (!san.isGood())
-                return false;
-        }
-        else
+        auto nodeIDit = nodeIDs.cbegin();
+        auto nodeDatait = data.begin();
+        TransactionStateSF filter(
+            mLedger->txMap().family().db(), app_.getLedgerMaster());
+
+        while (nodeIDit != nodeIDs.cend())
         {
-            san += mLedger->txMap().addKnownNode(
-                *nodeIDit, makeSlice(*nodeDatait), &filter);
-            if (!san.isGood())
-                return false;
-        }
+            if (nodeIDit->isRoot())
+            {
+                san += mLedger->txMap().addRootNode(
+                    SHAMapHash{mLedger->info().txHash},
+                    makeSlice(*nodeDatait),
+                    &filter);
+                if (!san.isGood())
+                    return false;
```

Comment:
Consider logging here and below as we do with an Account State node.
eg:
`
                if (!san.isGood())
                {
                    JLOG(m_journal.warn()) << "Unable to add TX root node";
                    return false;
                }
`